eternalcore-core/src/main/java/com/eternalcode/core/util/Preconditions.java

```diff
@@ -0,0 +1,59 @@
+package com.eternalcode.core.util;
+
+import java.util.Collection;
+import java.util.Map;
+
+public final class Preconditions {
+
+    private Preconditions() {
```

Comment:
Możesz także wyrzucić UnsupportedOperationException, tak jak robimy to w przypadku innych utility klas.